workers/data_refinery_workers/processors/illumina.R

```diff
@@ -0,0 +1,447 @@
+#suppressWarnings(data <- read.table(filePath, header=TRUE, fill=TRUE))
+
+# Check input paramters and parse out data we need
+if (probeIDColumn == ""){
+  message("Got no probID column...")
+  probeIDColumn <- "PROBE_ID"
+}
+
+if (!(probeIDColumn %in% colnames(data)))
+{
+  stop(paste0("No ", probeIDColumn, " column in the data file."))
+}
+
+probeIDs <- data[,probeIDColumn]
+
+
+if (length(exprColumns) == 0)
+  stop(paste0("No columns match this pattern: ", exprColumnPattern))
+
+exprData <- as.matrix(data[,exprColumns,drop=FALSE])
+rownames(exprData) <- probeIDs
+
+if (detectionPValueColumnPattern == "")
+  detectionPValueColumnPattern <- "Detection Pval"
+pValueColumns <- grep(detectionPValueColumnPattern, colnames(data), ignore.case=TRUE)
+
+if (length(pValueColumns) == 0)
+  stop(paste0("No columns match this pattern: ", detectionPValueColumnPattern))
+
+if (length(exprColumns) != length(pValueColumns))
+  stop(paste0("The number of expression columns [", length(exprColumns), "] did not match the number of p-value columns [", length(pValueColumns), "]."))
+
+pValueData <- as.matrix(data[,pValueColumns,drop=FALSE])
+rownames(pValueData) <- probeIDs
+
+# Extract platform-specific data
+if (platform == "illuminaHumanv2") {
+  probeSequenceRef <- illuminaHumanv2PROBESEQUENCE
+  probeQualityRef <- illuminaHumanv2PROBEQUALITY
+  #probeReporterRef <- illuminaHumanv2REPORTERGROUPID # This indicates control probe info
+  probeGeneRef <- illuminaHumanv2ENSEMBLREANNOTATED
+} else {
+  probeSequenceRef <- illuminaHumanv4PROBESEQUENCE
+  probeQualityRef <- illuminaHumanv4PROBEQUALITY
+  #probeReporterRef <- illuminaHumanv4REPORTERGROUPID # This indicates control probe info
+  probeGeneRef <- illuminaHumanv4ENSEMBLREANNOTATED
+}
+
+# Parse probe sequence info
+probeSequences <- as.data.frame(probeSequenceRef[mappedkeys(probeSequenceRef)])
+rownames(probeSequences) <- probeSequences[,1]
+
+# Limit to probes that overlap between data and probe sequences
+commonProbes <- intersect(rownames(exprData), rownames(probeSequences))
+if (length(commonProbes) < 1000)
+  stop("Not enough probes overlap between the annotations and the data.")
+
+exprData <- exprData[commonProbes,,drop=FALSE]
+pValueData <- pValueData[commonProbes,,drop=FALSE]
+probeSequences <- probeSequences[commonProbes,2]
+
+# Do normalization
+normData <- scanNorm(exprData, probeSequences, signalPValueData=pValueData, numCores=numCores)
+
+# Parse probe quality info
+#   (It makes sense to do this after normalization so the SCAN model has more data to work with)
+probeQuality <- as.data.frame(probeQualityRef[mappedkeys(probeQualityRef)])
+goodProbeIndices <- which(grepl("Good", probeQuality$ProbeQuality))
+perfectProbeIndices <- which(grepl("Perfect", probeQuality$ProbeQuality))
+probesToKeep <- probeQuality[c(goodProbeIndices, perfectProbeIndices),1]
+probesToKeep <- intersect(probesToKeep, rownames(normData))
+
+normData <- normData[probesToKeep,,drop=FALSE]
+
+# Map probes to genes
```

Comment:
We might want a [tidyverse](https://www.tidyverse.org) solution to this bit instead, but that has the disadvantage of additional dependencies. (The sizes of the `3.4.3` tagged versions of the [`rocker/r-ver`](https://hub.docker.com/r/rocker/r-ver/tags/) and [`rocker/tidyverse`](https://hub.docker.com/r/rocker/tidyverse/tags/) Docker images are 232 MB and 705 MB, respectively, for reference.)